tests/cases/py_client/cli_test.py

```diff
@@ -267,6 +268,17 @@ def testUploadDownload(self):
                          downloadDir), username='mylogin', password='password')
         self.assertEqual(ret['exitVal'], 0)
 
+        # Test uploading with reference
+        @httmock.urlmatch(netloc='localhost', path='/api/v1/file$', method='POST')
+        def checkParams(url, request):
+            query = six.moves.urllib.parse.parse_qs(url[3])
+            self.assertIn('reference', query)
+            self.assertIn('reference_string', query['reference'])
```

Comment:
This is good -- one thing I also like to do is add a sanity check to make sure these asserts are actually getting called, since we actually have no way of knowing based on this. So, what I do is create a mutable structure like a list outside the `checkParams` function, then inside `checkParams` do `my_list.append(query)`. Then, right below the `invokeCli` call, you can assert that the list has one entry, and move the current assertions down below that.